### PR TITLE
Upper-case python package name

### PR DESCRIPTION
**Pull request by grwlf**
Trivial rename: `pennylane-catalyst` -> `PennyLane-Catalyst`

[sc-52633]

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -24,7 +24,7 @@ USER $USERNAME
 # Install Python kernel for use with Jupyter Notebooks.
 RUN pip install --no-cache-dir ipykernel
 
-RUN pip install --no-cache-dir pennylane-catalyst
+RUN pip install --no-cache-dir PennyLane-Catalyst
```
Comment (josh146) [inline .devcontainer/Dockerfile:27]: @grwlf how come this line has changed?

File: doc/dev/installation.rst [hunk 2]
```diff
@@ -8,7 +8,7 @@ pre-built binaries are being distributed via the Python Package Index (PyPI) for
 
 .. code-block:: console
 
-    pip install pennylane-catalyst
+    pip install PennyLane-Catalyst
```
Comment (josh146) [inline doc/dev/installation.rst:11]: does this need to change? by convention most pip commands are lowercase

File: doc/dev/installation.rst [hunk 3]
```diff
@@ -203,7 +203,7 @@ You can also choose to build the custom Catalyst dialects only, with:
 Frontend
 """"""""
 
-To install the ``pennylane-catalyst`` Python package (the compiler frontend) in editable mode:
+To install the ``PennyLane-Catalyst`` Python package (the compiler frontend) in editable mode:
```
Comment (josh146) [inline doc/dev/installation.rst:206]: same here, I would prefer to leave this lowercase

File: README.md [hunk 4]
```diff
@@ -77,7 +77,7 @@ binaries are being distributed via the Python Package Index (PyPI) for Python ve
 higher. To install it, simply run the following ``pip`` command:
 
 ```console
-pip install pennylane-catalyst
+pip install PennyLane-Catalyst
```
Comment (josh146) [inline README.md:80]:
```suggestion
pip install pennyLane-catalyst
```

File: setup.py [hunk 5]
```diff
@@ -176,7 +176,7 @@ def run(self):
 # - `qjit`: Path to the JIT compiler decorator provided by the compiler.
 setup(
     classifiers=classifiers,
-    name="pennylane-catalyst",
+    name="PennyLane-Catalyst",
```
Comment (josh146) [inline setup.py:179]: @grwlf my understanding is that for this PR, _only_ this line should be changed, alongside any lines that refer to explicit wheel/egg files. But pip commands should remain as-is

Comment (grwlf) [reply setup.py:179]: Got it, I have reverted the changes other than this line and the egg-related ones

File: README.md (hunk 4 above)
Comment (grwlf) [reply README.md]: Reverted

File: doc/dev/installation.rst (hunk 3 above)
Comment (grwlf) [reply doc/dev/installation.rst]: Reverted

File: doc/dev/installation.rst (hunk 2 above)
Comment (grwlf) [reply doc/dev/installation.rst]: Reverted

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (grwlf) [reply .devcontainer/Dockerfile]: Reverted

**Pull request merged**